### PR TITLE
MVP reporting

### PR DESCRIPTION
**Pull request by kalouo**
## Issue 
 
#16 

## Overview

- implements creation of CSV file detailing all payments to delegators. 
- implements creation of HJSON summarizing the baker's income

## Changes

Creates a new `fs-client` folder with associated logic to create reports and write them to the file system. 

## Usage and Testing

Test locally.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@ node_modules/
 config.json
 config.hjson
 .env
-yarn.lock
\ No newline at end of file
+yarn.lock
+reports/
```
Comment (cryi) [inline .gitignore:6]: We want here: 
```
reports/*
!reports/.gitkeep
```
And add .gitkeep file into that folder.

File: src/engine/interfaces.ts [hunk 2]
```diff
@@ -18,7 +18,14 @@ export interface StepArguments {
   distributableRewards: BigNumber;
 }
 
+export enum EPaymentType {
+  Delegator = "delegator_payment",
+  FeeIncome = "fee_income_payment",
+  BondReward = "bond_reward",
+}
```
Comment (cryi) [inline src/engine/interfaces.ts:25]: What about nice names? (`Bond Reward`, `Fee Income Payment`...) This output is primarily for users so I would go with nice names if possible.

File: src/engine/steps/resolveExcludedPaymentsByMinimumAmount.ts [hunk 3]
```diff
@@ -16,7 +16,11 @@ export const resolveExcludedPaymentsByMinimumAmount = (
   const delegatorPayments = _.map(cycleReport.delegatorPayments, (payment) => {
     if (payment.amount.lt(minimumPaymentAmount)) {
       feeIncome = feeIncome.plus(payment.amount);
-      return { ...payment, amount: new BigNumber(0) };
+      return {
+        ...payment,
+        amount: new BigNumber(0),
+        note: "payment_below_minimum",
```
Comment (cryi) [inline src/engine/steps/resolveExcludedPaymentsByMinimumAmount.ts:22]: Same here I would put there `Payment below minimum`

File: src/fs-client/index.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+  { id: "total_baker_balance", title: "total_baker_balance" },
+  { id: "total_cycle_rewards", title: "total_cycle_rewards" },
+  { id: "fee_rate", title: "fee_rate" },
+  { id: "fee", title: "fee" },
+  { id: "amount", title: "amount" },
+  { id: "recipient", title: "recipient" },
+  { id: "tx_hash", title: "tx_hash" },
+  { id: "timestamp", title: "timestamp" },
+  { id: "note", title: "note" },
+];
```
Comment (cryi) [inline src/fs-client/index.ts:26]: Do we need the baker balance, total reward? I feel like these should be in the hjson report for the cycle. 
Not sure about fee rate and fee. Maybe join it into one column? (like `0.001 (5%)`) I feel like there is too many columns 😄 

And please move timestamp at the beginning. It should be first imo.

File: src/fs-client/index.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+  await writeHJSON(path, cycleReport.cycle, prepareCycleReport(cycleReport));
+};
+
+const prepareCycleReport = (cycleReport: CycleReport) => {
+  return {
+    cycle: cycleReport.cycle,
+    feeIncome: cycleReport.feeIncome.toString(),
+    lockedBondRewards: cycleReport.lockedBondRewards.toString(),
+  };
```
Comment (cryi) [inline src/fs-client/index.ts:58]: Adjust based on edits in delegator report.

**Review by cryi — CHANGES_REQUESTED** [review]

File: src/fs-client/index.ts (hunk 4 above)
Comment (kalouo) [reply src/fs-client/index.ts:26]: The vision here is to make things more easily auditable – e.g.  a delegator is able to calculate the payment amount based on cycle rewards, proportion of the delegated stake and fee rate.

Comment (cryi) [reply src/fs-client/index.ts:26]: I see, but still the `baker balance`, `total reward` fields feel redundant. Users can check baker report before calculating 🤔 Eventually we can provide command for displaying just the audible part on demand.

**Pull request merged**